WooCommerce/Classes/ViewRelated/Orders/Order Creation/CustomAmounts/AddCustomAmountViewModel.swift

```diff
@@ -0,0 +1,56 @@
+import Combine
+import WooFoundation
+import UIKit
+import SwiftUI
+
+typealias CustomAmountEntered = (_ amount: String, _ name: String) -> Void
+
+final class AddCustomAmountViewModel: ObservableObject {
+    let formattableAmountTextFieldViewModel: FormattableAmountTextFieldViewModel
+    let onCustomAmountEntered: CustomAmountEntered
+
+    init(locale: Locale = Locale.autoupdatingCurrent,
+         storeCurrencySettings: CurrencySettings = ServiceLocator.currencySettings,
+         onCustomAmountEntered: @escaping CustomAmountEntered) {
+        self.formattableAmountTextFieldViewModel = FormattableAmountTextFieldViewModel(locale: locale, storeCurrencySettings: storeCurrencySettings)
+        self.onCustomAmountEntered = onCustomAmountEntered
+        listenToAmountChanges()
+    }
+
+    /// Variable that holds the name of the custom amount.
+    ///
+    @Published var name = ""
+    @Published var shouldDisableDoneButton: Bool = true
+
+    var customAmountPlaceholder: String {
+        Localization.customAmountPlaceholder
+    }
+
+    func doneButtonPressed() {
+        let customAmountName = name.isNotEmpty ? name : customAmountPlaceholder
+        onCustomAmountEntered(formattableAmountTextFieldViewModel.amount, customAmountName)
+        reset()
+    }
+
+    func reset() {
+        name = ""
+        shouldDisableDoneButton = true
+
+        formattableAmountTextFieldViewModel.reset()
+    }
+}
+
+private extension AddCustomAmountViewModel {
+    func listenToAmountChanges() {
+        formattableAmountTextFieldViewModel.$amount.map { _ in
+            !self.formattableAmountTextFieldViewModel.amountIsValid
+        }.assign(to: &$shouldDisableDoneButton)
+    }
```

Comment:
We seem to do too many things here for the method name, perhaps we could split the method into `binding/listenToAmountChanges` and `validation`?